drivers/win7/driver.c

```diff
@@ -322,12 +322,51 @@ NTSTATUS _write_mmio_mem(PHYSICAL_ADDRESS pa, unsigned int len, void* pData)
         DbgPrint("[chipsec] ERROR: no space for mapping\n");
         return STATUS_UNSUCCESSFUL;
       }
-    if (len == 8) 
-      { 
-        count = 2; 
-      }
     DbgPrint("[chipsec] writing %u bytes to MMIO address 0x%08x_%08x (virtual = %#010x)", len, pa.HighPart, pa.LowPart, (UINTN)va);
-    WRITE_REGISTER_BUFFER_ULONG((volatile ULONG*)(va), (PULONG)pData, count);
+    
+    switch(len)
+    {
+      case 1:
+        WRITE_REGISTER_BUFFER_UCHAR((volatile UCHAR*)(va), (PUCHAR)pData, count);
+        break;
+      case 2:
+        WRITE_REGISTER_BUFFER_USHORT((volatile USHORT*)(va), (PUSHORT)pData, count);
+        break;
+      case 8:
+        count = 2;

```

Comment:
Add a comment here that the missing break is intentional and that it just sets up two DWORD writes.